### PR TITLE
+ Update to our non-deprecated use of observer.

### PR DESCRIPTION
**Pull request by lbwexler**
+ explicitly use mobx-react-lite, preparing for future removal of mobx-react

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X ] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: core/HoistComponent.js [hunk 1]
```diff
@@ -74,6 +74,11 @@ export function hoistComponent(config) {
         "The 'model' config passed to hoistComponent() is incorrectly specified: provide a spec returned by either uses() or creates()."
     );
 
+    warnIf(
+        !isMemo && isObserver,
+        'Cannot create an observer component without `memo`.  Memo is built-in to MobX observable. Component will be memoized.'
+    );
+
```
Comment (amcclain) [inline core/HoistComponent.js:81]: Is the memo doc comment still relevant / useful?

> Components that are known to be unable to make effective use of memo (e.g. container components) may set this to `false`.

I never really understood what this meant, to be honest 😄  Is it referring to the really low-level comps like vbox?

**Review by amcclain — APPROVED** [review]
Looks good - thanks for reviewing and improving on my change from yesterday. In particular the forwardRef stuff remains somewhat opaque to me, so I wasn't sure how to test that.

Two small questions to take or leave - happy to have this merged whenever you are ready to do so.

File: core/HoistComponent.js [hunk 2]
```diff
@@ -6,10 +6,10 @@
  */
 import {CreatesSpec, elemFactory, ModelPublishMode, ModelSpec, uses} from '@xh/hoist/core';
 import {useOwnedModelLinker} from '@xh/hoist/core/impl/UseOwnedModelLinker';
-import {throwIf, withDefault} from '@xh/hoist/utils/js';
+import {throwIf, warnIf, withDefault} from '@xh/hoist/utils/js';
 import classNames from 'classnames';
 import {isFunction, isPlainObject, isString} from 'lodash';
-import {observer} from 'mobx-react';
+import {observer} from 'mobx-react-lite';
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply core/HoistComponent.js:12]: yes regarding the backward compatibility and mobx-react.  Looking forward to simplifying thing by removing all that in v39.

great catch -- regarding the re-export --- will fix that up to come from the new mobx-react-light package!

File: core/HoistComponent.js (hunk 1 above)
Comment (lbwexler) [reply core/HoistComponent.js:81]: yeah -- still getting set in a large number of places -- main point is that if you are actually rendering children that *are passed to you*, you always get passed a new object references to those children, so memoization is always defeated.  Read that somewhere -- and understood it clearly at the time, but yeah its a tough model to really grok.

**Pull request merged**